src/library/scala/collection/MapView.scala

```diff
@@ -47,6 +49,25 @@ object MapView {
   /** A `MapOps` whose collection type and collection type constructor are (mostly) unknown */
   type SomeMapOps[K, +V] = MapOps[K, V, SomeIterableConstr, _]
 
+  def empty[K, V]: MapView[K, V] = EmptyMapView.asInstanceOf[MapView[K, V]]
+
+  @SerialVersionUID(3L)
+  private case object EmptyMapView extends MapView[Any, Nothing] {
+    override def get(key: Any): Option[Nothing] = None
+    override def iterator: Iterator[Nothing] = Iterator.empty
+    override def knownSize: Int = 0
+    override def isEmpty: Boolean = true
+  }
+
+  @SerialVersionUID(3L)
+  private class OverSeqView[K, +V](underlying: SeqView[(K, V)]) extends AbstractMapView[K, V] {
+    private[this] lazy val tupleView = underlying.reverse.distinctBy(_._1)
+    def get(key: K): Option[V] = tupleView.find(_._1 == key).map(_._2)
```

Comment:
I’m not sure we want this. Maps should have O(1) lookups. Otherwise the behavior of views will (again) become unpredictable.

Reply:
Also, `tupleView` evaluates all the elements of the underlying sequence, which defeats the purpose of going through a view…

Reply:
Should we basically do this then?

```
private lazy val _map: Map[K, V] = underlying.toMap
def get(key: K): Option[V] = _map.get(key)
```

Reply:
That would solve the `O(1)` issue but not the fact that this view would still evaluate all the elements of its underlying collections during the first lookup.

Reply:
I don't think it's really possible to avoid both right? Fundamentally, you have to at least get to the end of the IterableOnce before you can know what any Key-Value is

Reply:
Indeed, this is why I think we have to make `MapView[K, V]` extend `IterableOps[(K, V), View, View[(K, V)]]`

Reply:
Hmm, maybe as a compromise, it could be..

```scala
trait MapView[K, +V]
  extends View[(K, V)]
  with MapOps[K, V, ({ type l[X, Y] = View[(X, Y)] })#l, MapView[K, V]]
    with ViewOps[(K, V), View, MapView[K, V]] {
  ...
```

That way, the mapFactory produces `View[(K, V)]`, but all the operations described in the issue return `MapView`s.

Reply:
perhaps the `MapView#mapFactory` could specialize `from` for input of type `Map[K, V]`, in this case to return `MapView`

Thanks for the feedback